File: array/rotate_array/rotate_array.go
package main

import "fmt"


func rotate (arr []int, k int) []int {
	n := len(arr)
	k = k % n
	return append(arr[n-k:], arr[:n-k]...)
}

func main () {
	a := []int{1, 2, 3, 4, 5, 6, 7}
	fmt.Println("in: ", a)
	fmt.Println("expected: ", []int{5, 6, 7, 1, 2, 3, 4})
	fmt.Println("out: ", rotate(a, 3))
}

File: array/rotate_array/rotate_array.js
function rotate(arr, k) {
    const n = arr.length
    k = k % n
    return arr.slice(n - k).concat(arr.slice(0, n - k))
}


a = [1, 2, 3, 4, 5, 6, 7]
console.log("in: ", a)
console.log("expected: ", [5, 6, 7, 1, 2, 3, 4])
console.log("out: ", rotate(a, 3))